lib/config.module.ts

```diff
@@ -150,22 +150,22 @@ export class ConfigModule {
   private static loadEnvFile(
     options: ConfigModuleOptions,
   ): Record<string, any> {
-    try {
-      const envFilePath = options.envFilePath || resolve(process.cwd(), '.env');
-      const config = dotenv.parse(fs.readFileSync(envFilePath));
-      if (options.expandVariables) {
-        const configExpandedVars = {
-          parsed: config,
-        };
-        return dotenvExpand(configExpandedVars).parsed || {};
-      }
-      return config;
-    } catch (err) {
-      if (options.envFilePath || (err && err.code !== 'ENOENT')) {
-        throw err;
+    const envFilePaths = Array.isArray(options.envFilePath)
```

Comment:
Any specific reason why `try / catch` was removed? I am quite sure an error will still be thrown if one of the files doesn't exist.

Reply:
A `fs.existsSync(envFilePath)` check is introduced. So it'll not try to load files that does not exist.

The reason is that `*.env.local` files will not always be present (as it is not intended to be checked into git). It will only be created if the developer wants to override something locally.

Reply:
Ahh... makes sense. Missed that part, my bad! Looks good to me. Good job 👍 

The only thing to note is your IDE doesn't automatically new lines at the EOF, although it's not requirement here, but is a good practice. For reference, check your `.env.local` 😄 

Lastly, in the docs, please mention the precedence of loading the .env files. Usually `.env` is considered primary and then `.env.production`, but in this case, if `.env` is placed first in the array, it'll disrupt the order.

Reply:
> The only thing to note is your IDE doesn't automatically new lines at the EOF, although it's not requirement here, but is a good practice. For reference, check your `.env.local` :smile:

Well actually, I deliberately left out the newline at EOF of `.env.local`, cause I noticed the already existing `.env` file didn't have it. :sweat_smile: 

> Lastly, in the docs, please mention the precedence of loading the .env files. Usually `.env` is considered primary and then `.env.production`, but in this case, if `.env` is placed first in the array, it'll disrupt the order.

By docs, do you mean the `nestjs/docs.nestjs.com` repo? I already mentioned about the precedence: "If a variable is found in multiple files, the first one takes precedence."

https://github.com/nestjs/docs.nestjs.com/pull/1043/files#diff-c541c05c0821982446cbd0c8e126731fR63

Reply:
> Well actually, I deliberately left out the newline at EOF of .env.local, cause I noticed the already existing .env file didn't have it. 😅

Yeah, I believe those files were created through the console / terminal & not the IDE. 

> By docs, do you mean the nestjs/docs.nestjs.com repo? I already mentioned about the precedence: "If a variable is found in multiple files, the first one takes precedence."

Makes sense. 👍

Reply:
Added newlines at EOF for those files.